enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,501 @@
+Support for the ExternalDNS CRD Source should be added to the ExternalDNS Operator in OCP 4.10.
+* Support Gateway APIs in ExternalDNS/ExternalDNS Operator. See [upstream issue 2045](https://github.com/kubernetes-sigs/external-dns/issues/2045)
+for more context. Support for Gateway APIs should be added after the initial "tech-preview" ExternalDNS Operator is available in OCP 4.9
+* Perform a gap analysis between ExternalDNS Operator and the [Ingress Operator's DNS provider implementations](https://github.com/openshift/cluster-ingress-operator/tree/master/pkg/dns).
+This could be performed as a part of the ExternalDNS Operator work for OCP 4.10.
+
+## Proposal
+
+The proposal is based on experience gathered and work accomplished with existing OpenShift operators.
+
+The following tasks would be completed as a part of this enhancement:
+
+* Create a GitHub repository to host the operator source code in OpenShift's GitHub org,
+with the hopes of moving the operator's source code into an upstream k8s-sigs repository in the future.
+* Leverage popular frameworks and libraries, e.g. controller-runtime and kubebuilder, to simplify development
+of the operator and align with upstream communities.
+* Manage dependencies through [Go Modules](https://blog.golang.org/using-go-modules).
+* Create user and developer documentation for running ExternalDNS operator (on both OpenShift and vanilla Kubernetes).
+* Create tooling to simplify building, running, testing, etc. the operator.
+* Create tests that reduce bugs in the code and allow for continuous integration (tests should be compatible
+with OpenShift CI, in addition to vanilla Kubernetes environments running in public clouds).
+* Integrate the operator with the OpenShift toolchain, e.g. [openshift/release](https://github.com/openshift/release),
+which would ultimately be required for productization.
+* Provide manifests for quickly deploying the operator (including a minimal ExternalDNS Operator CR manifest).
+* Create tooling to simplify ExternalDNS troubleshooting. For example, provide detailed information via `status.conditions` and `status.relatedObjects`
+in the externalDNS `clusteroperator` resource when running on OpenShift.
+* Integrate ExternalDNS with the OpenShift monitoring toolchain. For example, add
+[metrics support](https://github.com/openshift/cluster-monitoring-operator) for the operator and its operand(s).
+
+
+In addition, the following functionality is expected as part of the operator (note that this is not an exhaustive list):
+
+* Introduce a CRD to be used by the operator to manage ExternalDNS and its dependencies.
+* Expand upon the existing OpenShift [ingressoperator/dnsrecord](https://github.com/openshift/api/blob/master/operatoringress/v1/types.go)
+CRD to implement the ExternalDNS base [CRD source](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/contributing/crd-source.md).
```

Comment:
We moved this to non-goals (for 4.9), so we need to be clear here that we are not committing to making this change as part of this specific enhancement.

Reply:
Sure. I added a note here that this is work for post 4.9.